src/btree/bt_delete.c

```diff
@@ -207,6 +207,8 @@ __wti_delete_page(WT_SESSION_IMPL *session, WT_REF *ref, bool *skipp)
 
     /* Set the page to its new state. */
     WT_REF_SET_STATE(ref, WT_REF_DELETED);
+    __wt_atomic_addv16(&ref->ref_changes, 1);
```

Comment:
I think we should increase this number before we mark the ref deleted. It is the general principle for all cases that we increase this value before any real change to the ref. Otherwise, we may in a race condition that We change the ref and we context switch before we increase the counter.